### PR TITLE
Fix csvdiff script

### PR DESCRIPTION
**Pull request by milljm**
Fixed a few discovered issues with csvdiff.py

Removed old no-longer-used csvdiffer unittest and added a new set of unittests, specific to our stand-alone csvdiff.py script.

Added documentation.

Removed the deprecation message when using override tolerances functionality, as we want to continue to support it within test spec files.

Moved the csvdiff.py script, to moose/scripts. As it is a 'more' friendly location users are already familiar with.

Small un-related fix to the TestHarness documentation for the justification formatting of some of the tables.

### CODE REVIEW

**Comment by milljm** [comment]
I am still finding small bugs with this. Please do not merge.

File: scripts/csvdiff.py [hunk 1]
```diff
@@ -84,41 +84,39 @@ def parseComparisonFile(self, config_file):
...
+                # Possible global header containing tollerance params
+                if not re.match(r'^\s', a_line) and words.intersection(tollerance_params):
+                    custom_params['RELATIVE'] = self.getParamValues(words.intersection(tollerance_params).pop(), a_line)[0]
+
+                # Possible field containing floor params
+                if field_key and words.intersection(zero_params):
+                    custom_params['FIELDS'][field_key[0]]['ZERO'] = self.getParamValues(words.intersection(zero_params).pop(), a_line)[0]
+
+                # Possible field containing tollerance params
+                if field_key and words.intersection(tollerance_params):
```
Comment (permcody) [inline scripts/csvdiff.py]: tollerance -> tolerance

File: modules/doc/content/python/CSVDiff.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    z                    # min: 1.000e+00 @ t1          max: 1.000e+02 @ t0
+```
+
+We can also instruct CSVDiff to ignore field 'y' all together:
+
+```
+TIME STEPS relative 1 floor 0  # min: 0 @ t0  max: 1 @ t1
+
+GLOBAL VARIABLES relative 5.5e-06 floor 1e-11
+    !y                   # min: 0.000e+00 @ t0          max: 1.000e-06 @ t1
```
Comment (permcody) [inline modules/doc/content/python/CSVDiff.md]: Wow - is this something that's supported by Exodiff?

File: test/tests/test_harness/gold/b_csvdiff_comparison_out.csv [hunk 3]
```diff
@@ -0,0 +1,7 @@
+time,u_field
```
Comment (permcody) [inline test/tests/test_harness/gold/b_csvdiff_comparison_out.csv]: I'm not sure I like your override tests. If you only have one field to override, it's equivalent to a global override. Can you perhaps add at least one more column to these tests to verify that they actually behave differently than the global versions?

**Review by permcody — CHANGES_REQUESTED** [review]
Looking really good! Your override tests need a bit more work. Also I'm a little confused on some of the behavior. `--only-compare-custom` is suppose to only compare the columns listed in `--custom-columns`, However, the doc string for custom-columns seems to indicate that that's the default behavior. So which is it? Does --custom-columns behave as an override for specific columns but also compare the remaining ones with the global, or does it only compare the custom? 

If we go with the way Exodiff works, listing columns in the comp file makes it so ONLY those values get compared. This is why the `-summary` option populates the skeleton file with a list of all columns so you can simply remove the ones you want to ignore. With that in mind, I don't even think you need the `--only-compare-custom` option at all! That should be the only mode for this tool. 

Part of what you are doing is making the tool we want. Don't worry about making it pass the Frankenstein tests that are checked in. Each of those tests just needs to be updated appropriately. This should simplify the logic a lot!

**Comment by milljm** [comment]
I am sure there are some options we can remove entirely... for sure.
I'll do another pass over how all that argument logic is suppose to work. As I recall, I remember seeing that option (only-do-custom-column) and feeling un-easy about it.

File: modules/doc/content/python/CSVDiff.md (hunk 2 above)
Comment (milljm) [reply modules/doc/content/python/CSVDiff.md]: Hmmm I am not sure. I'll test it. If it does not work, I suppose I should remove this feature from csvdiff?

Or, just update our statement about both summery files being interchangeable...

**Comment by permcody** [comment]
Ben just put in a PR to remove some of those. I might have one test that uses the other option, but it should also just die.

**Comment by permcody** [comment]
Yes, please remove it, if it breaks the testing, Ben and I can look at it and submit a separate PR or just add a commit to this one.

**Comment by permcody** [comment]
@milljm - are you still working on this PR?

**Comment by milljm** [comment]
@permcody Nope... ready to be re-reviewed...

File: modules/doc/content/python/CSVDiff.md (hunk 2 above)
Comment (permcody) [reply modules/doc/content/python/CSVDiff.md]: What did you find out about this syntax? Does this exist in Exodiff? @bwspenc - do you know?

Comment (milljm) [reply modules/doc/content/python/CSVDiff.md]: I think it does work. Here is my custom comparison file:
```
[#]> cat simple.cmp
COORDINATES absolute 1.e-6    # min separation not calculated
TIME STEPS relative 1.e-6 floor 0.0     # min:               0 @ t1 max:               1 @ t2
# No GLOBAL VARIABLES
NODAL VARIABLES relative 1.e-6 floor 0.0
	!u  # min:               0 @ t1,n1	max:               1 @ t2,n21
# No ELEMENT VARIABLES
# No NODESET VARIABLES
# No SIDESET VARIABLES
```

And the results when I ask exodiff to use it:
```
[#]> exodiff -f  simple.cmp gold/simple_diffusion_out.e simple_diffusion_out.e 
<... trimmed ...>
Nodal coordinates will be compared .. tol:    1e-06 (absolute), floor:        0
Time step values will be compared .. tol:    1e-06 (relative), floor:        0
No Global variables on either file.
Nodal variables will not be compared. <--- THIS IS STATED
No Element variables on either file.
No Element Attribute variables on either file.
No Nodeset variables on either file.
No Sideset variables on either file.
<... trimmed ...>

exodiff: Files are the same
```

As opposed to this when I don't:
```
<... trimmed ...>
Nodal coordinates will be compared .. tol:    1e-06 (absolute), floor:        0
Time step values will be compared .. tol:    1e-06 (relative), floor:        0
No Global variables on either file.
Nodal variables to be compared:
	u                                tol:    1e-06 (relative), floor:        0
No Element variables on either file.
No Element Attribute variables on either file.
No Nodeset variables on either file.
No Sideset variables on either file.
Sideset Distribution Factors will be compared .. tol:    1e-06 (relative), floor:        0
<... trimmed ...>
exodiff: Files are the same
```

Comment (permcody) [reply modules/doc/content/python/CSVDiff.md]: OK!

**Review by permcody — DISMISSED** [review]

**Comment by permcody** [comment]
@jason - will you rebase and repush? There are a lot of failures on this PR that I believe are all related to all of the issues we've had this week. I'm hoping a rebase will clean up the testing.

**Review by permcody — APPROVED** [review]

**Pull request merged**